### PR TITLE
Fix and update spider habit_burger

### PR DESCRIPTION
**Pull request by abbensid2**

### CODE REVIEW

File: locations/spiders/habit_burger.py [hunk 1]
```diff
@@ -1,80 +1,32 @@
...
-                phone = phone.strip() if phone else None
-                opening_hours = Selector(text=poi).xpath('//div[@class="lochrs"]/text()').extract()
-                opening_hours = opening_hours = (
-                    ", ".join([hours.strip() for hours in opening_hours]) if opening_hours else None
+    sitemap_urls = ["https://www.habitburger.com/locations-sitemap.xml"]
+    sitemap_rules = [("", "parse_sd")]
+    wanted_types = ["Restaurant"]
+
+    def inspect_item(self, item, response):
+        openingHour = json.loads(response.xpath('//*[@id="content"]/script[1]/text()').get())
```
Comment (Cj-Malone) [inline locations/spiders/habit_burger.py:17]: This is to get the ld json? If so doing it in `pre_process_data` or `post_process_item` will already have `ld_data` available.

Comment (Cj-Malone) [reply locations/spiders/habit_burger.py:17]: I've not checked, but the first example I looked at was valid ld format, but with a none default time format, we need a better solution, but for now the following may work:
`oh.from_linked_data(ld_data, time_format="%I:%M%p")`

Comment (abbensid2) [reply locations/spiders/habit_burger.py:17]: @Cj-Malone  
You need an update here, you want me to

```
        elif linked_data.get("openingHours"):
            rules = linked_data["openingHours"]
            if not isinstance(rules, list):
                rules = re.findall(
                    r"((\w{2,3}|\w{2,3}\s?\-\s?\w{2,3}|(\w{2,3},)+\w{2,3})\s(\d\d:\d\d)\s?\-\s?(\d\d:\d\d))",
                    rules,
                )
                rules = [r[0] for r in rules]

            for rule in rules:
                days, time_ranges = rule.split(" ", 1)

                if time_ranges.lower() == "closed":
                    continue

                for time_range in time_ranges.split(","):
                    start_time, end_time = time_range.split("-")

                    start_time = start_time.strip()
                    end_time = end_time.strip()

                    if "-" in days:
                        start_day, end_day = days.split("-")

                        start_day = sanitise_day(start_day)
                        end_day = sanitise_day(end_day)

                        for day in day_range(start_day, end_day):
                            self.add_range(day, start_time, end_time, time_format)
                    else:
                        for day in days.split(","):
                            self.add_range(day, start_time, end_time, time_format)
```

**Comment by abbensid2** [comment]
@Cj-Malone 
I don't like to touch the common part,
I make it work with `from_linked_data`

**Comment by Cj-Malone** [comment]
Would you mind putting a couple examples in #4241, then it's on my TODO list.

**Pull request merged**